plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/usecase/CacheGlobalAddonsGroups.kt

```diff
@@ -0,0 +1,29 @@
+package org.wordpress.android.fluxc.usecase
+
+import org.wordpress.android.fluxc.network.rest.wpcom.wc.addons.dto.AddOnGroupDto
+import org.wordpress.android.fluxc.persistence.dao.AddonsDao
+import org.wordpress.android.fluxc.persistence.mappers.toAddonEntity
+import org.wordpress.android.fluxc.persistence.mappers.toAddonGroupEntity
+import org.wordpress.android.fluxc.persistence.mappers.toAddonOptionEntity
+import javax.inject.Inject
+
+internal class CacheGlobalAddonsGroups @Inject constructor(private val addonsDao: AddonsDao) {
+    suspend fun invoke(
+        globalAddonGroups: List<AddOnGroupDto>,
+        remoteSiteId: Long
+    ) {
+        addonsDao.deleteGlobalAddonsForSite(remoteSiteId)
+
+        globalAddonGroups.forEach { group ->
+            val globalAddonGroupEntityId = addonsDao.insertGroup(group.toAddonGroupEntity(remoteSiteId))
+            group.addons.forEach { addon ->
+                val addonEntity = addon.toAddonEntity(globalAddonGroupEntityId)
+                val addonEntityId = addonsDao.insertAddons(addonEntity)
+                addon.options?.forEach { addonOption ->
+                    val addonOptionEntity = addonOption.toAddonOptionEntity(addonLocalId = addonEntityId)
+                    addonsDao.insertAddonOptions(addonOptionEntity)
+                }
+            }
+        }
+    }
+}
```

Comment:
I was wondering if maybe we could extract into functions each loop operation to improve readability and model type operation. WDYT?

```suggestion
internal class CacheGlobalAddonsGroups @Inject constructor(private val addonsDao: AddonsDao) {
    suspend fun invoke(
        globalAddonGroups: List<AddOnGroupDto>,
        remoteSiteId: Long
    ) {
        addonsDao.deleteGlobalAddonsForSite(remoteSiteId)
        
        globalAddonGroups.forEach { group ->
            addonsDao.insertGroup(group.toAddonGroupEntity(remoteSiteId))
                    .let { group.cacheAddons(it) }
        }
    }

    private suspend fun AddOnGroupDto.cacheAddons(groupID: Long) {
        addons.forEach { addon ->
            addon.toAddonEntity(groupID)
                    .let { addonsDao.insertAddons(it) }
                    .let { addon.cacheAddonOptions(it) }
        }
    }

    private suspend fun WCProductAddonModel.cacheAddonOptions(addonID: Long) {
        options?.map { it.toAddonOptionEntity(addonID) }
                ?.forEach { addonsDao.insertAddonOptions(it) }
    }
}
```

Reply:
Good suggestion, definitely improves readability! I've changed it a little, I hope that form is okay too: 41ea70f3ff39e7d08ae21d7d490239ae3218199a

Reply:
Looks great @wzieba, thanks for applying the suggestion!